src/drivers/cmakeDriver.ts

```diff
@@ -1286,7 +1297,7 @@ export abstract class CMakeDriver implements vscode.Disposable {
             if (this.useCMakePresets) {
                 telemetryProperties = {
                     CMakeExecutableVersion: cmakeVersion ? util.versionToString(cmakeVersion) : '',
-                    CMakeGenerator: this.getGeneratorNameForTelemetry(),
+                    CMakeGenerator: (trigger === ConfigureTrigger.taskProvider) ? taskPreset?.generator || "" : this.getGeneratorNameForTelemetry(),
```

Comment:
We don't want to skip the PII scrub in the generator name for telemetry. You will need to send the override preset to `getGeneratorNameForTelemetry`

Reply:
when we use kits and there is no generator defined, we use `getGeneratorNameForTelemetry` to get the best generator, but when we are using the presets, if the generator entry is empty, we only log a message and won't find the best.

https://github.com/microsoft/vscode-cmake-tools/blob/9a9b1a51c582b0f93cc10e0d083915f55d4495db/src/drivers/cmakeDriver.ts#L459

I can send a specific message in telemetry instead of `""`, for example, `"no.generator"`, would that be good?

Reply:
My concern is about sending `presetOverride?.generator` directly to telemetry.  That value needs to be scrubbed for PII first.